messaging/src/test/java/org/axonframework/eventhandling/ReplayTokenTest.java

```diff
@@ -135,4 +154,1470 @@ void createReplayTokenReturnsReplayToken() {
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(10, emptySet());
+            TrackingToken result = replayToken.advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result);
+            assertTrue(ReplayToken.isReplay(result), "Event at exact reset position should be a replay");
+        }
+
+        @Test
+        void eventAfterResetPositionWithoutGapsEndsReplay() {
+            // tokenAtReset at index 10, no gaps
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, emptySet());
+            ReplayToken replayToken = (ReplayToken) ReplayToken.createReplayToken(tokenAtReset, null);
+
+            // Event at index 11 - after reset position
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(11, emptySet());
+            TrackingToken result = replayToken.advancedTo(newToken);
+
+            assertFalse(ReplayToken.isReplay(result), "Event after reset position should not be a replay");
+        }
+
+        @Test
+        void eventInGapOfTokenAtResetIsNotReplay() {
+            // tokenAtReset at index 10 with gaps at 7,8 - processor saw 0-6,9,10 before reset
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+            ReplayToken replayToken = (ReplayToken) ReplayToken.createReplayToken(tokenAtReset, null);
+
+            // Event at index 7 - this was a gap, never processed before reset
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(7, emptySet());
+            TrackingToken result = replayToken.advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result, "Should still be in replay mode");
+            assertFalse(ReplayToken.isReplay(result),
+                    "Event that was in gap of tokenAtReset should NOT be marked as replay");
+        }
+
+        @Test
+        void eventAfterGapButBeforeResetIndexIsReplay() {
+            // tokenAtReset at index 10 with gaps at 7,8 - processor saw 0-6,9,10 before reset
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+            ReplayToken replayToken = (ReplayToken) ReplayToken.createReplayToken(tokenAtReset, null);
+
+            // Event at index 9 - after the gaps but before reset index, WAS processed before
+            // Note: newToken has gaps 7,8 because they weren't seen yet during this replay
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(9, setOf(7L, 8L));
+            TrackingToken result = replayToken.advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result);
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event at index 9 was processed before reset, should be a replay");
+        }
+
+        // EXAMPLE: forgot about events that were gaps - tokenAtReset NEVER covers the newToken
+        // IS IT VALID SCENARIO? I have null token at the beginning and then advance to 9
```

Comment:
Yes, this may happen, depending on how the `GapAwareTrackingToken` is constructed by the `JpaEventStorageEngine` or `JdbcEventStorageEngine`, based on the gap configuration options to both stores. So, you can remove the comment on this method.

```suggestion
```